presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/StaticMetastoreLocator.java

```diff
@@ -76,17 +85,15 @@ public StaticMetastoreLocator(List<URI> metastoreUris, @Nullable String metastor
     public ThriftMetastoreClient createMetastoreClient(Optional<String> delegationToken)
             throws TException
     {
-        List<HostAndPort> metastores = new ArrayList<>(addresses);
-        Collections.shuffle(metastores.subList(1, metastores.size()));
+        List<Integer> indices = backoffs.stream()
+                .sorted(Comparator.comparingLong(backoff -> backoff.getBackoffDuration()))
+                .map(backoffs::indexOf)
+                .collect(toImmutableList());
 
         TException lastException = null;
-        for (HostAndPort metastore : metastores) {
+        for (int index : indices) {
             try {
-                ThriftMetastoreClient client = clientFactory.create(metastore, delegationToken);
-                if (!isNullOrEmpty(metastoreUsername)) {
-                    client.setUGI(metastoreUsername);
-                }
```

Comment:
restore this `if` block here

Reply:
the logic moved to `getClient` function.